Application to utilize the created ng2C3 component

sudharsanam opened the pull request:
- Application written using WebPack and Angular 2
- Currently able to render the created line chart using the application
- All the steps are mentioned in Read me document

ToDO:
- Mentioning template URL in component does not work. Need to look at the loaders in webpack for HTML to do.

/cc @arunkumars08 @mageshwaranr

arunkumars08 commented:
For Readme.md file. We can worry about the proper structure later. For now at least make it like points that comes in each line. So it becomes easier to read.

arunkumars08 commented on app/app.component.ts:
```diff
@@ -0,0 +1,26 @@
+import {Component, CUSTOM_ELEMENTS_SCHEMA} from '@angular/core';
+
+//Including Ng2C3 component into the application
+import {Ng2C3} from './library/ng2C3.component';
```
I don't think it is 'ng2C3.component' instead it should be 'ng2c3.component'

arunkumars08 replied:
Did it work for you? If yes, I would be amazed

sudharsanam replied:
It did work !

arunkumars08 commented on README.md:
```diff
@@ -1,8 +1,19 @@
+          For creating a Line Chart
+          Create a Wrapper and include ng2-c3 component 
+          <div style="width:500px;height:500px">
+              <ng2-c3 bind-data="_data"></ng2-c3>
```
Why is there bind-data used here? I am just curious. I am guessing if it is made as a component, we can simply use [data]="_data". I have seen bind being used in directives. Just have a look.

sudharsanam replied:
Any advantages of using it in that way ?

arunkumars08 replied:
If we go by the looks, we can first avoid unnecessary word ' bind'. The component way looks cleaner. Advantages we can worry later on.

arunkumars08 commented on app/app.module.ts:
```diff
@@ -0,0 +1,13 @@
+import {NgModule}      from '@angular/core';
+import {BrowserModule} from '@angular/platform-browser';
+import {AppComponent}  from './app.component';
+import {Ng2C3} from './library/ng2C3.component';
```
C shouldn't be in capital in ng2C3.component

arunkumars08 commented on app/library/ng2c3.component.ts:
```diff
@@ -74,7 +75,7 @@ export class Ng2C3 {
     try {
       this.__render( this.data );
-    } catch(err : Error) {
+    } catch(err) {
```
If not the strict type, at least 'any' can be mentioned here I guess

sudharsanam replied:
It is augggested not to use any type for error catching ! Hence removed intentionally !

arunkumars08 replied:
Lemme check more on that. For now I was able to find that closely relates to this, https://github.com/Microsoft/TypeScript/issues/298

arunkumars08 replied:
For now we can ignore this. We can think about this later.

arunkumars08 commented on app/app.component.ts:
```diff
@@ -0,0 +1,26 @@
+
+@Component({
+    selector: 'my-app',
+    template: '<div style="width:500px;height:500px"><ng2-c3 bind-data="_data"></ng2-c3></div>'
```
Single quotes in template works fine. In the future if somebody needs to add more template taking it to many lines, they might miss the quotes as multi-line quotes are "`". I think it would be better to always use this.

sudharsanam replied:
True. Let's add it to the guidelines so that we can follow the same way ..

arunkumars08 commented on app/app.component.ts:
```diff
@@ -0,0 +1,26 @@
+            columns: [
+            ['data1', 30, 200, 100, 400, 150, 250],
+            ['data2', 50, 20, 10, 40, 15, 25]
+        ]
```
Type 'bar' has to be given for 'Barchart'

sudharsanam replied:
This is just for line chart and not for bar chart

arunkumars08 replied:
Ok sorry. I overlooked. My bad

sudharsanam commented:
@arunkumars08 : 
 I have addressed all the review comments. I have tried to make readme.md into points. If you have any specific ways in mind, go ahead and modify it and raise a PR. we shall follow the approach which is good. As far as typing of error is concerned, My IDE clearly indicates not to put any typing for it. 
 Re-review this once again hence this can be pushed!
